### ISSUE TITLE
Estonian language support

### ISSUE THREAD

**Issue by mangotree123** [sspanak/tt9#688]
Hi! I've been using tt9 for a while and it's fantastic. Any way support for Estonian could be added? 
Thanks!

**Comment by sspanak** [comment]
Of course. If you'd like to help, you can give me a hand with searching a good word list/dictionary. If there is a respected university or academy, or institute that regulates the language and has issued something like "the big dictionary of Estonian", I could make a use of it. Such dictionaries are spell-checked and contain a lot of different word forms which results in very high quality suggestions. Since it is difficult for me to search in Estonian websites, I would appreciate if you could do this for me.

**Comment by mangotree123** [comment]
Hi, sorry for the late response, I was away from home for the holidays. The official Estonian dictionary is ÕS, but I can't seem to find a downloadable version of it. The official way to access it online is through https://sonaveeb.ee but it functions like an online dictionary.

**Comment by sspanak** [comment]
> Hi, sorry for the late response, I was away from home for the holidays.

All good!

> The official Estonian dictionary is ÕS, but I can't seem to find a downloadable version of it. The official way to access it online is through https://sonaveeb.ee/ but it functions like an online dictionary.

They seem to have [an API](https://github.com/keeleinstituut/ekilex/wiki/Ekilex-API). I can try using the `/api/word/details/{wordId}` endpoint. It provides a lot of nice info, including the word frequency. I am just not sure if I will be able to register and get a key from https://ekilex.ee/register

**Comment by mangotree123** [comment]
It seems like they don't limit access much, at least registering is as simple as any account creation process.
Not quite so sure about the API access, but it seems pretty straightforward too. 
If needed, I could try to make the request on your behalf.

**Comment by sspanak** [comment]
Sure, I don't mind getting a hand. 

The [sample API response](https://gist.github.com/mlaubre/876034582cbc51c54a133e79747a32d7) looks quite complex. It would be great if you could identify which parts of it contain words, word variants, synonyms and whatnot, but not foreign words. I'd prefer collecting as many as possible and then, if they happen to be too many, say above 1 million, I can easily filter them by frequency of use.

**Comment by sspanak** [comment]
@mangotree123, I was able to extract the words and added Estonian to TT9. I will include it in the next version. But before that, I would appreciate if you test it and let me know if anything is missing or if you notice any misspelled words. Here is [an APK](https://github.com/sspanak/tt9/releases/download/v1.0-inital/tt9-v45.0+et.apk) for you to try it out.
